topdown/eval_test.go

```diff
@@ -239,3 +243,100 @@ func TestContainsNestedRefOrCall(t *testing.T) {
 		})
 	}
 }
+
+func TestTopdownVirtualCacheFunctions(t *testing.T) {
+	ctx := context.Background()
+	store := inmem.New()
+	compiler := compileModules([]string{
```

Comment:
nit: I've found (over time) that it's better to move these kinds of fixtures into the test case definition. While it may create a bit of duplication, I find it much easier to reason about in the future (esp. when the fixture grows organically over time to accommodate new cases.)

E.g., I'd expand the cases to include a `module` field that will be compiled for each test case.

Reply:
Good point. Some of those organically grown test fixtures are hard to work with indeed.